### DISCUSSION TITLE
What is FaceSwap under the hood?

### DISCUSSION THREAD

**Discussion by amirshachar** [lllyasviel/Fooocus#1641] [General]
I'm curious if the Fooocus FaceSwap feature in the UI is basically a LoRA, ControlNet or something else in Stable Diffusion. How do I replicate the FaceSwap setup in ComfyUI, for example.

**Comment by mashb1t** [comment]
Fooocus uses the IP adapter ip-adapter-plus-face_sdxl_vit-h.bin, see https://github.com/lllyasviel/Fooocus/blob/main/modules/config.py#L492C70-L492C105

A ready-to-use implementation for ComfyUI can be found here: https://github.com/cubiq/ComfyUI_IPAdapter_plus

Hope this helps!

**Comment by amirshachar** [comment]
Brilliant - thank you.

**Comment by martinenkoEduard** [comment]
I wanted to ask the same question))
Already answered)
